documentation/asciidoc/computers/linux_kernel/building.adoc

```diff
@@ -1,342 +1,462 @@
+$ KERNEL=kernel8
+$ make ARCH=arm64 CROSS_COMPILE=aarch64-linux-gnu- bcm2711_defconfig
 ----
+| Raspberry Pi 3+
+| Raspberry Pi 4
+| Raspberry Pi 400
+| Raspberry Pi Zero 2 W
+| Raspberry Pi Compute Modules 3
+| Raspberry Pi Compute Module 4
+| Raspberry Pi Compute Module 4S
 
-For Raspberry Pi 5:
-
-[,bash]
+| Raspberry Pi 5
+.1+.^|
+[source,console]
 ----
-cd linux
-KERNEL=kernel_2712
-make ARCH=arm64 CROSS_COMPILE=aarch64-linux-gnu- bcm2712_defconfig
+$ cd linux
+$ KERNEL=kernel_2712
+$ make ARCH=arm64 CROSS_COMPILE=aarch64-linux-gnu- bcm2712_defconfig
 ----
+|===
 
-NOTE: The standard, `bcm2711_defconfig`-based kernel (`kernel8.img`) also runs on Raspberry Pi 5. For best performance you should use `kernel_2712.img`, but for situations where a 4KB page size is required then `kernel8.img` (`kernel=kernel8.img`) should be used.
-
-===== Build with configs
-
-NOTE: To speed up compilation on multiprocessor systems, and get some improvement on single-processor devices, use `-j n`, where n is the number of processors × 1.5. You can use the `nproc` command to see how many processors you have. 
+==== Build with modules and Device Tree
 
-====== For all 32-bit builds
+For 64-bit kernels, pass the image modules and DTB files like so:
 
-[,bash]
+[source,console]
 ----
-make ARCH=arm CROSS_COMPILE=arm-linux-gnueabihf- zImage modules dtbs
+$ make ARCH=arm64 CROSS_COMPILE=aarch64-linux-gnu- Image modules dtbs
 ----
 
-====== For all 64-bit builds
+For 32-bit kernels, pass the image modules and DTB files like so, prefixing each image module with `z`:
 
-NOTE: Note the difference between Image target between 32- and 64-bit.
-
-[,bash]
+[source,console]
 ----
-make ARCH=arm64 CROSS_COMPILE=aarch64-linux-gnu- Image modules dtbs
+$ make ARCH=arm CROSS_COMPILE=arm-linux-gnueabihf- zImage modules dtbs
 ----
 
-==== Install directly onto the SD card
+==== Install the kernel
 
-Having built the kernel, you need to copy it onto your Raspberry Pi and install the modules. This is best done directly using an SD card reader.
+Having built the kernel, you need to copy it onto your Raspberry Pi boot media (likely an SD card or SSD) and install the modules.
 
-First, use `lsblk` before and after plugging in your SD card to identify it. You should end up with something a lot like this:
+First, run `lsblk`. Then, connect your boot media. Run `lsblk` again; the new device represents your boot media. You should see output similar to the following:
 
 ----
 sdb
    sdb1
    sdb2
 ----
 
-with `sdb1` being the `FAT` filesystem (boot) partition, and `sdb2` being the `ext4` filesystem (root) partition.
+If `sdb` represents your boot media, `sdb1` represents the the `FAT32`-formatted **boot partition** and `sdb2` represents the (likely `ext4`-formatted) **root partition**.
 
-Mount these first, adjusting the partition letter as necessary:
+First, mount these partitions as `/mnt/boot` and `/mnt/root`, adjusting the partition letter to match the location of your boot media:
```

Comment:
It's mounting as just `mnt/boot` and `mnt/root`, not `/mnt/boot` and `/mnt/root` :stuck_out_tongue_winking_eye: 
(i.e. a `mnt` directory in the current folder, not the `/mnt` directory at the filesystem root - I assume that this is done to prevent any conflicts in case there's something else already mounted at `/mnt/boot` or `/mnt/root`)